components/service/glean/src/main/java/mozilla/components/service/glean/private/TimingDistributionMetricType.kt

```diff
@@ -33,26 +34,65 @@ data class TimingDistributionMetricType(
     private val logger = Logger("glean/TimingDistributionMetricType")
 
     /**
-     * Accumulates a sample value to the timing distribution.
+     * Start tracking time for the provided metric and associated object. This
+     * records an error if it’s already tracking time (i.e. start was already
+     * called with no corresponding [stopAndAccumulate]): in that case the original
+     * start time will be preserved.
      *
-     * @param sample This is the sample to accumulate
+     * @param anyObject The object to associate with this timing.  This allows
+     * for concurrent timing of events associated with different objects to the
+     * same timespan metric.
      */
-    fun accumulate(sample: Long) {
+    fun start(anyObject: Any) {
```

Comment:
nit: I would be a bit more explicit with the name of the parameter here. Maybe something like `timerId` instead of `anyObject`